ogr/ogrsf_frmts/miramon/ogrmiramonlayer.cpp

```diff
@@ -0,0 +1,2613 @@
+                        pszString, MM_MAX_LON_FIELD_NAME_DBF);
+                    CPLFree(pszString);
+                }
+                else
+                {
+                    MM_strnzcpy(
+                        phMiraMonLayer->pLayerDB->pFields[iField].pszFieldName,
+                        poFeatureDefn->GetFieldDefn(iField)->GetNameRef(),
+                        MM_MAX_LON_FIELD_NAME_DBF);
+                }
+            }
+
+            if (poFeatureDefn->GetFieldDefn(iField)->GetAlternativeNameRef())
+            {
+                if (phMiraMonLayer->nCharSet != MM_JOC_CARAC_UTF8_DBF)
+                {
+                    char *pszString =
+                        CPLRecode(poFeatureDefn->GetFieldDefn(iField)
+                                      ->GetAlternativeNameRef(),
+                                  CPL_ENC_UTF8, CPL_ENC_ISO8859_1);
+                    MM_strnzcpy(phMiraMonLayer->pLayerDB->pFields[iField]
+                                    .pszFieldDescription,
+                                pszString, MM_MAX_BYTES_FIELD_DESC);
+                    CPLFree(pszString);
+                }
+                else
+                {
+                    MM_strnzcpy(phMiraMonLayer->pLayerDB->pFields[iField]
+                                    .pszFieldDescription,
+                                poFeatureDefn->GetFieldDefn(iField)
+                                    ->GetAlternativeNameRef(),
+                                MM_MAX_BYTES_FIELD_DESC);
+                }
+            }
+            phMiraMonLayer->pLayerDB->nNFields++;
+        }
+    }
+
+    CPLDebug("MiraMon", "Fields to MiraMon translated.");
+    return OGRERR_NONE;
+}
+
+/****************************************************************************/
+/*                       TranslateFieldsValuesToMM()                        */
+/*                                                                          */
+/*      Translase ogr Fields to a structure that MiraMon can understand     */
+/****************************************************************************/
+
+OGRErr OGRMiraMonLayer::TranslateFieldsValuesToMM(OGRFeature *poFeature)
+
+{
+    if (poFeatureDefn->GetFieldCount() == 0)
+    {
+        // MiraMon have private DataBase records
+        hMMFeature.nNumMRecords = 1;
+        return OGRERR_NONE;
+    }
+
+    MM_EXT_DBF_N_MULTIPLE_RECORDS nIRecord;
+    int nNumFields = poFeatureDefn->GetFieldCount();
+    MM_EXT_DBF_N_MULTIPLE_RECORDS nNumRecords;
+    hMMFeature.nNumMRecords = 0;
+
+    for (int iField = 0; iField < nNumFields; iField++)
+    {
+        OGRFieldType eFType = poFeatureDefn->GetFieldDefn(iField)->GetType();
+        const char *pszRawValue = poFeature->GetFieldAsString(iField);
+
+        if (eFType == OFTStringList)
+        {
+            char **panValues = poFeature->GetFieldAsStringList(iField);
+            nNumRecords = CSLCount(panValues);
+            if (nNumRecords == 0)
+                nNumRecords++;
```

Comment:
that seems dangerous to me as you are likely going to use panValues[nIRecord] = panValues[0] = nullptr below. Are you sure this is needed an safe?

Reply:
You're right. What I need in case of an empty stringlist at least one empty record. But I have not access to this one. So, I resize (and set to empty) but I don't use it. If I did it is because at some point I needed that. But it's better like that:
 nRealNumRecords = nNumRecords = CSLCount(papszValues);
and use nRealNumRecords in the for for (nIRecord = 0; nIRecord < nRealNumRecords; nIRecord++)

If nRealNumRecords == 0 then I resize to hMMFeature.nNumMRecords = max_function(hMMFeature.nNumMRecords, nNumRecords); but I fill that as empty